oat-sa/extension-tao-proctoring
# Feature/pr 520/remove kv table

**Ticket**
https://oat-sa.atlassian.net/browse/PR-520

**Description**
This PR aims to transcript proctoring improvements to latest version with possibility to configure. 
it includes:
- [Refactor to improve database access](https://oat-sa.atlassian.net/browse/PR-471) (https://github.com/oat-sa/extension-tao-proctoring/pull/892)
- [Script to migrate data](https://oat-sa.atlassian.net/browse/PR-495) ()
- [Script to update to new impl](https://oat-sa.atlassian.net/browse/PR-495) (https://github.com/oat-sa/extension-tao-proctoring/pull/895)

Also event listeners have been moved from DeliveryService to a new MonitoringListener.

**To test**
All these scenarios should be run with psql and mysql.
- Checkout this PR, install the plaftorm with this PR, pass a proctored DX
- Do not checkout this PR, install, pass a proctored DX then checkout this PR, update, and run the scripts
```
php index.php '\oat\taoProctoring\scripts\tools\MonitoringExtraFieldMigration' -h
```
You can still pass proctored DX
